src/FSharp.Core/fslib-extra-pervasives.fsi

```diff
@@ -509,6 +509,15 @@ namespace Microsoft.FSharp.Core.CompilerServices
         /// <returns></returns>
         abstract ResolveTypeName : typeName: string -> Type
 
+    [<RequireQualifiedAccess; StructuralEquality; StructuralComparison>]
+    type TypeProviderDiagnosticSeverity =
+        | Warning
+        | Error
+
+    [<AllowNullLiteral>]
```

Comment:
Why AllowNullLiteral?

Reply:
No good reason. I'm using `null` in case the TP is used in a old compiler, which does not support the context. I'll change it to an option.